ffi/capi/src/errors.rs

```diff
@@ -66,13 +66,13 @@ pub mod ffi {
 
     #[derive(Debug, PartialEq, Eq)]
     #[repr(C)]
-    #[diplomat::rust_link(icu::calendar::FromStrError, Enum, compact)]
+    #[diplomat::rust_link(icu::calendar::ParseError, Enum, compact)]
     #[cfg(any(
         feature = "icu_datetime",
         feature = "icu_timezone",
         feature = "icu_calendar"
     ))]
-    pub enum CalendarFromStrError {
+    pub enum CalendarParseError {
```

Comment:
I get how this name came to be, but it's still weird because we're not parsing calendars. Actually the `CalendarError` above should probably be called `DateTimeError`, as you get this when constructing a datetime fails. You get this one when parsing a datetime or date or time fails, so I think `DateTimeParseError` would be better. But this can be a follow-up to fix the other name as well.

Reply:
This all comes back to the `icu_datetime` vs `icu_calendar` crate name. The `icu_datetime` crate is formatting for datetimes, so that is the correct name for it. The `icu_calendar` crate's primary value proposition, and the vast majority of core it exports, involves non-Gregorian calendars. It also happens to export the `Date` and `Time` types that we use in `icu_datetime` for formatting.

By our current naming conventions, `DateTimeParseError` would be an error originating from parsing a string in the `icu_datetime` crate, but that is not what this error is.

`icu_calendar` is more of a utility crate, except it's not, because it depends heavily on CLDR data and algorithms. But maybe if we go with `cldr_pattern` and/or `cldr_unit`, we could rename `icu_calendar` to `cldr_datetime` or something like that.

Reply:
> By our current naming conventions, DateTimeParseError would be an error originating from parsing a string in the icu_datetime crate, but that is not what this error is.

Errors are not prefixed by crates anymore, errors are named for the operations where they originate.

Reply:
Oh?

- `LocaleParseError` == `icu::locale::ParseError`
- `FixedDecimalParseError` == `fixed_decimal::ParseError`
- `FixedDecimalLimitError` == `fixed_decimal::LimitError`
- `TimeZoneInvalidOffsetError` == `icu::timezone::InvalidOffsetError`

...

Reply:
- `LocaleParseError` == `icu4x::Locale::from_string`' error
- `FixedDecimalParseError` == `icu4x::FixedDecimal::from_string`' error
- `FixedDecimalLimitError` == `icu4x::FixedDecimal::from_double`'s error
- `TimeZoneInvalidOffsetError` == `icu4x::CustomTimeZone::try_set_gmt_offset_seconds`' error

...

Reply:
https://github.com/unicode-org/icu4x/issues/5274